packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -193,18 +202,52 @@ export class BucketDeployment extends CoreConstruct {
       throw new Error('Distribution must be specified if distribution paths are specified');
     }
 
+    if (props.useEfs && !props.vpc) {
+      throw new Error('Vpc must be specified if useEfs is set');
+    }
+
+    const accessPointPath = '/lambda';
+    let accessPoint;
+    if (props.useEfs && props.vpc) {
+      const accessMode = '0777';
+      const fileSystem = this.getOrCreateEfsFileSystem(scope, {
+        vpc: props.vpc,
+        removalPolicy: cdk.RemovalPolicy.DESTROY,
+      });
+      accessPoint = fileSystem.addAccessPoint('AccessPoint', {
+        path: accessPointPath,
+        createAcl: {
+          ownerUid: '1001',
+          ownerGid: '1001',
+          permissions: accessMode,
+        },
+        posixUser: {
+          uid: '1001',
+          gid: '1001',
+        },
+      });
```

Comment:
The access point should depend on the mount targets (this is likely also why you'd get deletion problems if you create the VPC inline)...


```suggestion
      accessPoint = fileSystem.addAccessPoint('AccessPoint', {
        path: accessPointPath,
        createAcl: {
          ownerUid: '1001',
          ownerGid: '1001',
          permissions: accessMode,
        },
        posixUser: {
          uid: '1001',
          gid: '1001',
        },
      });
      accessPoint.node.addDependency(fileSystem.mountTargetsAvailable);
```

Reply:
The stack deletion fails when we create VPC inline since VPC will be deleted first upon stack deletion. This deletes the VPCInternetGateway and after that lambda function fails with `Connect Timeout`. It then results in stack deletion failure.

![Screenshot 2021-07-27 at 12 30 12 PM](https://user-images.githubusercontent.com/10937004/127110949-339d71fa-726a-4c75-be99-9892b4eec405.png)
![Screenshot 2021-07-27 at 12 32 41 PM](https://user-images.githubusercontent.com/10937004/127110970-f10de1b7-234a-4c10-bb98-48ed78e26862.png)

Reply:
@keshav0891 I think what @RomainMuller is saying is that is you add that dependency, it will create a dependency between the lambda function and the VPC, which will make the VPC be deleted only after the lambda, avoiding the problem you describe. 

Have you tried it?

Reply:
I can try but i was thinking if it is right to take stack dependency on input arguments. We are making our lambda depend on input VPC which might be created in different stack. 
Also, this problem arises when VPC and lambda are in same stack and we want to delete the stack. I am not sure if it is a problem worth solving because our infrastructure stack is mostly different and the problem may not occur.

Let me know your thoughts here.

Reply:
@keshav0891 I don't see a problem with creating a dependency on the VPC. If its in the same stack, all is well, if its in a different stack, CDK still knows to detect those dependencies and delete the stacks in the correct order.

You're right that a VPC is in another stack many times, but if the app is small enough it can very well live in the same stack. The fix seems simple and I don't see a reason why not to apply it.

Reply:
I tried above suggestion with below code but still got similar error.
```
if (props.vpc) {
  handler.node.addDependency(props.vpc);
}
```
![Screenshot 2021-09-13 at 4 57 14 PM](https://user-images.githubusercontent.com/10937004/133080339-886bb451-dede-4e40-a953-65e0bba68aef.png)
![Screenshot 2021-09-13 at 5 31 08 PM](https://user-images.githubusercontent.com/10937004/133080364-0b4b747a-552e-47eb-a1f1-8d4fbed77f70.png)

Were you also suggesting similar change or am i missing something?

Reply:
The integration test for which it failed is as below
```
new s3deploy.BucketDeployment(this, 'DeployMeWithEfsStorageInlineVpc', {
  sources: [s3deploy.Source.asset(path.join(__dirname, 'my-website'))],
  destinationBucket,
  destinationKeyPrefix: 'efsInline/',
  useEfs: true,
  vpc: new ec2.Vpc(this, 'inlineVPC'),
  retainOnDelete: false, // default is true, which will block the integration test cleanup
});
```

Reply:
You mention you tried:

```js
if (props.vpc) { handler.node.addDependency(props.vpc); }
```

But have you tried @RomainMuller suggestion to the letter? i.e

```ts
accessPoint.node.addDependency(fileSystem.mountTargetsAvailable);
```

?

Reply:
Yes. I tried below code earlier and still got same issue.
```
accessPoint.node.addDependency(fileSystem.mountTargetsAvailable);
```

Reply:
Fixed with [commit](https://github.com/aws/aws-cdk/pull/15220/commits/f83f9283e498bab326ce984cbf2e0a69d429ae12). The trick was to make `BucketDeployment` depend on `VPC`.